cmd/fluxctl/release_cmd.go

```diff
@@ -179,7 +187,68 @@ func (opts *controllerReleaseOpts) RunE(cmd *cobra.Command, args []string) error
 
 		opts.dryRun = false
 	}
-	return await(ctx, cmd.OutOrStdout(), cmd.OutOrStderr(), opts.API, jobID, !opts.dryRun, opts.verbosity)
+
+	err = await(ctx, cmd.OutOrStdout(), cmd.OutOrStderr(), opts.API, jobID, !opts.dryRun, opts.verbosity)
+	if !opts.watch || err != nil {
+		return err
+	}
+
+	fmt.Fprintf(cmd.OutOrStderr(), "Monitoring rollout ...\n")
+	for {
+		completed := 0
+		services, err := opts.API.ListServicesWithOptions(ctx, v11.ListServicesOptions{Services: result.Result.AffectedResources()})
+		if err != nil {
+			return err
+		}
+
+		for _, service := range services {
+			writeRolloutStatus(service, opts.verbosity)
+
+			if service.Status == cluster.StatusReady {
+				completed++
+			}
+
+			if service.Rollout.Messages != nil {
+				fmt.Fprintf(cmd.OutOrStderr(), "There was a problem releasing %s:\n", service.ID)
+				for _, msg := range service.Rollout.Messages {
+					fmt.Fprintf(cmd.OutOrStderr(), "%s\n", msg)
+				}
+				return nil
+			}
+		}
+
+		if completed == len(services) {
+			fmt.Fprintf(cmd.OutOrStderr(), "All controllers ready.\n")
+			return nil
+		}
+
+		time.Sleep(2000 * time.Millisecond)
+	}
+}
+
+func writeRolloutStatus(service v6.ControllerStatus, verbosity int) {
+	w := newTabwriter()
+	fmt.Fprintf(w, "CONTROLLER\tCONTAINER\tIMAGE\tRELEASE\tREPLICAS\n")
+
+	if len(service.Containers) > 0 {
+		c := service.Containers[0]
+		fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%d/%d", service.ID, c.Name, c.Current.ID, service.Status, service.Rollout.Updated, service.Rollout.Desired)
+		if verbosity > 0 {
+			fmt.Fprintf(w, " (%d outdated, %d updated)", service.Rollout.Outdated, service.Rollout.Updated)
+		}
+		fmt.Fprintf(w, "\n")
+		for _, c := range service.Containers[1:] {
+			fmt.Fprintf(w, "\t%s\t%s\t\t\n", c.Name, c.Current.ID)
+		}
+	} else {
+		fmt.Fprintf(w, "%s\t\t\t%s\t%d/%d", service.ID, service.Status, service.Rollout.Ready, service.Rollout.Desired)
```

Comment:
should use `Updated` here too instead of `Ready`.

Reply:
Thanks Roli :tomato: